runtime/src/bank.rs

```diff
@@ -15178,4 +15178,41 @@ pub(crate) mod tests {
             Some(Vec::<TransactionLogInfo>::new()),
         );
     }
+
+    /// Test exceeding the accounts data budget by creating accounts in a loop
+    #[test]
+    fn test_accounts_data_budget_exceeded() {
+        use solana_program_runtime::accounts_data_meter::MAX_ACCOUNTS_DATA_LEN;
+        use solana_sdk::system_instruction::MAX_PERMITTED_DATA_LENGTH;
+
+        solana_logger::setup();
+        let (genesis_config, mint_keypair) = create_genesis_config(1_000_000_000_000);
+        let mut bank = Bank::new_for_tests(&genesis_config);
+        bank.activate_feature(&solana_sdk::feature_set::cap_accounts_data_len::id());
+
+        let result = loop {
```

Comment:
I would suggest using a `for _ in 0..x` loop here, else the test won't be able to fail properly as it won't halt.

Reply:
Fixed!

I keep the `loop` so that I can `break result` (which is not possible with a for-loop), and added an assert with a loop counter and maximum to ensure the test (1) is bounded, and (2) completes at the expected loop count.